### ISSUE TITLE
MacOS: parenthesis selection of code in the help browser is misbehaving

### ISSUE THREAD

**Issue by tremblap** [supercollider/supercollider#4849]
## Environment

* SuperCollider version: 3.11.0
* Operating system:MacOS

## Steps to reproduce
Load this code in a help document, or download the package at https://www.flucoma.org/download/

```
// load some buffers
(
b = Buffer.read(s,File.realpath(FluidBufCompose.class.filenameSymbol).dirname.withTrailingSlash ++ "../AudioFiles/Tremblay-AaS-SynthTwoVoices-M.wav");
c = Buffer.read(s,File.realpath(FluidBufCompose.class.filenameSymbol).dirname.withTrailingSlash ++ "../AudioFiles/Tremblay-SA-UprightPianoPedalWide.wav");
d = Buffer.new(s);
)

// with basic params (basic summing of each full buffer in all dimensions)
FluidBufCompose.process(s, source: b,  destination: d);
FluidBufCompose.process(s, source: c,  destination: d, destGain: 1.0);
d.query;
d.play;
```

## Expected vs. actual behavior

This code behaves as expected in a .scd document: you command-enter with the cursor anywhere between the parenthesis, or double-click either parenthesis, and only what is between the parenthesis gets executed (command-enter) or selected (doubleclick)

when the same code is in a .schelp file. and that file is opened in the help browser, the only way to get the expected behaviour is to doubleclick on the closing parenthesis. The other 2 behaviour (doubleclicking on the opening and command-enter) are both selecting the whole CODE:: section.

**Comment by brianlheim** [comment]
thanks @tremblap . when posting reproducers, please avoid things that depend on third party packages. it is much easier for you to edit your example than for maintainers to download an external package just to test this

**Comment by brianlheim** [comment]
i don't understand this ticket. on macOS, for me neither command-clicking nor double-clicking evaluates code. in both the help browser and in an SC document:
- double clicking on either paren selects the region between the parens
- double clicking inside the region selects a character
- command-click just moves the cursor

i've also never heard of this as a feature. can someone please explain how this is intended to work on macOS?

**Comment by brianlheim** [comment]
one thing i do notice. the following lines have this behavior if you double-click the top paren:

```supercollider
(
() // fine
"" // fine
("") // causes the rest of the doc from the opening paren onward to be selected
)
```

but, is the event of a double click something that we even intercept and handle? again, not aware that we were doing anything in particular in response to that event

**Comment by tremblap** [comment]
> please avoid things that depend on third party
sorry for that. For a help file I did not know how to send it otherwise since you need to load it as a library...

> i've also never heard of this as a feature.
sorry again for this, I thought I was clear. The 4 beahviours I get in this code when NOT in a schelp are:
1. doubleclicking on the top parenthesis selects everything between that point and its closing one
2. doubleclicking on the bottom parenthesis selects everything between that point and the top one
3. putting the cursor anywhere between the 2, and command-enter, selects and execute the code

When the same code is in a schelp, I lose 1 and 3, only 2 works

**Comment by brianlheim** [comment]
> For a help file I did not know how to send it otherwise since you need to load it as a library...

two better options are to either edit your example to something that anyone with a basic installation of SC can run, or refer to an existing example in documentation. it's not a big deal though, i wrote my own reproducer which i've shared above.

> When the same code is in a schelp, I lose 1 and 3, only 2 works

okay, thank you for clarifying. could you please edit the original post for clarity so other readers can more easily understand the ticket? it still says "command-click" and implies that double-clicking executes code.

your points 1 and 3 actually do not select the entire `CODE::` section on my system. the comment in the first line is unselected, but everything below it is. if that is what you mean, then AFAICT we are now describing the same behavior, which is: whenever you try to get the 'whole region' in a help document, double-quotes inside parens (and possibly other sequences of tokens) somehow muck it up. does that seem consistent with what you're seeing?

**Comment by tremblap** [comment]
> whenever you try to get the 'whole region' in a help document, double-quotes inside parens (and possibly other sequences of tokens) somehow muck it up. does that seem consistent with what you're seeing?

I'm not sure this is only that, but it does certain happen. What is bizarre is that double-clicking on the closing parenthesis parses the selection properly in the helpfiles...


>  could you please edit the original post

will do my best to de-clutter but I still don't know what are all the cases in which this happens.

**Comment by elgiano** [comment]
> but, is the event of a double click something that we even intercept and handle? again, not aware that we were doing anything in particular in response to that event

I just encountered 'dblclick' doing parenthesis matching in editor.js:
https://github.com/supercollider/supercollider/blob/40cf0177feb4e9316c95163dfaaa3134de1961c2/HelpSource/editor.js#L45

I'm not actually working on this issue, but since I saw it I thought it could have been helpful to report it here :)

**Comment by brianlheim** [comment]
fixed in #5044, thanks @elgiano !
